### PR TITLE
Improve UT coverage in cloud/scope package for PowerVS

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Signed-off-by: Prajyot-Parab <prajyot.parab2@ibm.com>

**What this PR does / why we need it**:
- Improve UT coverage in cloud/scope package for PowerVS

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Improve UT coverage in cloud/scope package for PowerVS
```

### CODE REVIEW

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: cloud/scope/powervs_image_test.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
```
[the remark this answers is not in the document]
Comment (Prajyot-Parab) [reply cloud/scope/powervs_image_test.go]: Fixed.

File: cloud/scope/powervs_machine_test.go [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog/v2/klogr"
+	infrav1beta1 "sigs.k8s.io/cluster-api-provider-ibmcloud/api/v1beta1"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/pkg/cloud/services/powervs/mock"
```
[the remark this answers is not in the document]
Comment (Prajyot-Parab) [reply cloud/scope/powervs_machine_test.go]: Fixed.

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**